tensorflow/compiler/xla/service/gpu/buffer_comparator.cc

```diff
@@ -405,11 +619,13 @@ StatusOr<bool> HostCompare(se::Stream* stream, se::DeviceMemoryBase lhs,
 
   const auto canonicalize = [](ComparisonType a) -> ComparisonType {
     if (std::is_same<ElementType, Eigen::half>::value && a) {
-      constexpr ComparisonType kMaxFp16Value = 65505.;
+      constexpr ComparisonType kMaxFp16Value =
+          std::is_same<ElementType, Eigen::half>::value ? 65505. : 0;
```

Comment:
Aren't you already inside the branch which guarantees you that `ElementType` is Eigen::half?

Reply:
I agree, but compiler still errors about overflow.  I added the following comments on this in the previous PR:

This change is needed, because 65505 can't fit in int8. Refactoring this lambda to a set of template/overloaded helper functions is another solution.

Reply:
My bad.  The error was caused by a wrong comparison type.  With comparison type being float, this lambda doesn't have to change.  Reverting the changes.